blivet/tasks/availability.py

```diff
@@ -246,15 +285,108 @@ def available_resource(name):
-BLOCKDEV_MDRAID_PLUGIN = blockdev_plugin("mdraid")
-BLOCKDEV_MPATH_PLUGIN = blockdev_plugin("mpath")
-BLOCKDEV_SWAP_PLUGIN = blockdev_plugin("swap")
+# libblockdev btrfs plugin required technologies and modes
+BLOCKDEV_BTRFS_ALL_MODES = (blockdev.BtrfsTechMode.CREATE |
+                            blockdev.BtrfsTechMode.DELETE |
+                            blockdev.BtrfsTechMode.MODIFY |
+                            blockdev.BtrfsTechMode.QUERY)
+BLOCKDEV_BTRFS = BlockDevTechInfo(plugin_name="btrfs",
+                                  check_fn=blockdev.btrfs_is_tech_avail,
+                                  technologies={blockdev.BtrfsTech.MULTI_DEV: BLOCKDEV_BTRFS_ALL_MODES,
+                                                blockdev.BtrfsTech.SUBVOL: BLOCKDEV_BTRFS_ALL_MODES,
+                                                blockdev.BtrfsTech.SNAPSHOT: BLOCKDEV_BTRFS_ALL_MODES})
+BLOCKDEV_BTRFS_TECH = BlockDevMethod(BLOCKDEV_BTRFS)
+
+# libblockdev crypto plugin required technologies and modes
+BLOCKDEV_CRYPTO_ALL_MODES = (blockdev.CryptoTechMode.CREATE |
+                             blockdev.CryptoTechMode.OPEN_CLOSE |
+                             blockdev.CryptoTechMode.QUERY |
+                             blockdev.CryptoTechMode.ADD_KEY |
+                             blockdev.CryptoTechMode.REMOVE_KEY |
+                             blockdev.CryptoTechMode.RESIZE)
+BLOCKDEV_CRYPTO = BlockDevTechInfo(plugin_name="crypto",
+                                   check_fn=blockdev.crypto_is_tech_avail,
+                                   technologies={blockdev.CryptoTech.LUKS: BLOCKDEV_CRYPTO_ALL_MODES,
+                                                 blockdev.CryptoTech.ESCROW: blockdev.CryptoTechMode.CREATE})
+BLOCKDEV_CRYPTO_TECH = BlockDevMethod(BLOCKDEV_CRYPTO)
+
+# libblockdev dm plugin required technologies and modes
+BLOCKDEV_DM_ALL_MODES = (blockdev.DMTechMode.CREATE_ACTIVATE |
+                         blockdev.DMTechMode.REMOVE_DEACTIVATE |
+                         blockdev.DMTechMode.QUERY)
+BLOCKDEV_DM = BlockDevTechInfo(plugin_name="dm",
+                               check_fn=blockdev.dm_is_tech_avail,
+                               technologies={blockdev.DMTech.MAP: BLOCKDEV_DM_ALL_MODES,
+                                             blockdev.DMTech.RAID: BLOCKDEV_DM_ALL_MODES})
+BLOCKDEV_DM_TECH = BlockDevMethod(BLOCKDEV_DM)
+
+# libblockdev loop plugin required technologies and modes
+BLOCKDEV_LOOP_ALL_MODES = (blockdev.LoopTechMode.CREATE |
+                           blockdev.LoopTechMode.CREATE |
+                           blockdev.LoopTechMode.DESTROY |
+                           blockdev.LoopTechMode.MODIFY |
+                           blockdev.LoopTechMode.QUERY)
+BLOCKDEV_LOOP = BlockDevTechInfo(plugin_name="loop",
+                                 check_fn=blockdev.loop_is_tech_avail,
+                                 technologies={blockdev.LoopTech.LOOP_TECH_LOOP: BLOCKDEV_LOOP_ALL_MODES})  # FIXME
```

Comment:
What is it to be fixed here?

Reply:
That was a reminder for me to try to fix the constant name (`blockdev.LoopTech.LOOP_TECH_LOOP`) in libblockdev.